adam/ontology/__init__.py

```diff
@@ -179,3 +190,81 @@ class OntologyProperty:
 
     def __repr__(self) -> str:
         return f"+{self._handle}"
+
+
+@attrs(frozen=True, slots=True, repr=False)
+class HierarchicalObjectSchema:
+    """
+    A representation of objects made of `SubObject` via `SubObjectRelation`
+
+    The Hierarchical Objects allow for the representation of complex objects as more individual
+    parts. For example a Person has a head, torso, left arm, right arm, left leg, right leg as
+    parts of a whole relationship. The schema also contains the relationship the subobjects have
+    with each other.
+    """
+
+    parent_object: OntologyNode = attrib(validator=instance_of(OntologyNode))
+    """
+    The top-level `OntologyNode` which this Schema represents
+    """
+    sub_objects: ImmutableSet["SubObject"] = attrib(
+        converter=_to_immutableset, default=immutableset()
+    )
+    """
+    A set of `SubObject` which form parts of the whole. SubObjects themselves are a hierarchy
+    """
+    sub_object_relations: ImmutableSet["SubObjectRelation"] = attrib(
+        converter=_to_immutableset, default=immutableset()
+    )
+    """
+    A set of `SubObjectRelation` which define how the `SubObject` relate to one another. These 
+    relationships are considered "true at time of perception" but not always true because a Person 
+    can move their arm either above or below their head.
+    
+    Does this need to be addressed?
+    """
+
+
+@attrs(frozen=True, slots=True, repr=False)
+class SubObject:
+    """
+    A `HierarchicalObjectSchema` which is a part of a whole.
```

Comment:
Well, this isn't itself a `HierarchicalObjectSchema`; rather it *has* a `HierarchicalObjectSchema`. Perhaps better would be:

```
A sub-component of a generic type of object.

This is for use only in constructing `HierarchicalObjectSchema`\ ta.
```